test/integration/deprecated_specs.cc

```diff
@@ -38,3 +42,89 @@ TEST(DeprecatedSpecs, Spec1_2)
   sdf::SDFPtr sdf(new sdf::SDF());
   EXPECT_FALSE(sdf::initFile(filename, sdf));
 }
+
+////////////////////////////////////////////////////
+TEST(DeprecatedElements, CanEmitErrors)
+{
+  sdf::SDFPtr sdf(new sdf::SDF());
+  sdf::init(sdf);
+  auto elem = std::make_shared<sdf::Element>();
+  elem->SetRequired("-1");
+  elem->SetName("testElem");
+  // Change the root of sdf so we can test "testElem" in isolation
+  sdf->Root(elem);
+  auto config = sdf::ParserConfig::GlobalConfig();
+  config.SetWarningsPolicy(sdf::EnforcementPolicy::ERR);
+  sdf::Errors errors;
+  EXPECT_TRUE(sdf::readString(
+      "<sdf version='1.8'><testElem/></sdf>", config, sdf, errors));
+  ASSERT_FALSE(errors.empty());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_DEPRECATED, errors[0].Code());
+}
+
+////////////////////////////////////////////////////
+TEST(DeprecatedElements, CanEmitWarningWithErrorEnforcmentPolicy)
+{
+  // Capture sdfwarn output, which is treamed to std::cerr
+  std::stringstream buffer;
+  auto old = std::cerr.rdbuf(buffer.rdbuf());
+
+#ifdef _WIN32
+  sdf::Console::Instance()->SetQuiet(false);
+#endif
+
+  auto contains = [](const std::string &_a, const std::string &_b)
```

Comment:
nit Benefit as lambda not immediately clearer.
Can you comment as to why? (e.g. "`EXPECT_PRED2` does not functions, so we define as lambda"?)
Also, is my speculation true? (if not, can you just make it a function?)

Reply:
Just being lazy. Changed to function in https://github.com/osrf/sdformat/pull/543/commits/9592c002b72c1e9d187b4487857e37b92b1d0aec

Reply:
Excellent!